src/main/java/com/google/devtools/build/lib/bazel/bzlmod/BzlmodRepoRuleCreator.java

```diff
@@ -79,6 +81,10 @@ public static Rule createRule(
       // This literally cannot happen -- we just created the package!
       throw new IllegalStateException(e);
     }
+    if (rule.containsErrors()) {
+      throw Starlark.errorf(
+          "failed to instantiate '%s' from this module extension", ruleClass.getName());
```

Comment:
this is not exactly accurate -- this method can be called when we're creating repos from Bazel modules too. I think just "failed to instantiate repo '%s'" is probably good enough.